File: csharp/src/detectors/sensitive-information-leak/SensitiveInformationLeak.cs
using System.Net.Security;
using System.Security.Cryptography.Xml;

namespace SensitiveInformation
{
    class SensitiveInformationLeak
    {
        // {fact rule=sensitive-information-leak@v1.0 defects=1}
        public void SensitiveInformationLeakNoncompliant()
        {
            string url = "http://api.example.com";
            string apiKey = "YOUR_API_KEY";
            using (HttpClient client = new HttpClient())
            {
                // Noncompliant: Send sensitive data over HTTP connection
                HttpResponseMessage response = client.GetAsync(url + "?api_key=" + apiKey).Result;
            }
        }
        // {/fact}
        
        // {fact rule=sensitive-information-leak@v1.0 defects=0}
        public void SensitiveInformationLeakCompliant()
        {
            string url = "http://api.example.com";
            string apiKey = "YOUR_API_KEY";
            using (HttpClient client = new HttpClient())
            {
                // Noncompliant: Do not send sensitive data over HTTP connection
                HttpResponseMessage response = client.GetAsync(url).Result;
            }
        }
        // {/fact}
    }
}

File: ruby/src/detectors/sensitive-information-leak/sensitive-information-leak.rb
=begin
Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
SPDX-License-Identifier: Apache-2.0
=end

# {fact rule=sensitive-information-leak@v1.0 defects=1}
def sensitive_information_leak_noncompliant
    # Noncompliant: User-controlled data is passed in find.
    @user = User.find(params[:id])

    respond_to do |format|
      format.html 
      format.json { render :json => @user }
    end
end
# {/fact}

# {fact rule=sensitive-information-leak@v1.0 defects=0}
def sensitive_information_leak_compliant
    # Compliant: Argument in find is not user-controlled.
    @user = User.find(session[:id])

    respond_to do |format|
      format.html 
      format.json { render :json => @user }
    end
end
# {/fact}
